src/core/prometheus.cc

```diff
@@ -655,6 +657,25 @@ class metrics_handler : public handler_base  {
         }
         return false;
     }
+    std::function<bool(const mi::labels_type&)> make_filter(const httpd::request& req) {
+        std::unordered_map<sstring, std::regex> matcher;
+        auto labels = mi::get_local_impl()->get_labels();
+        for (auto&& qp : req.query_parameters) {
+            if (labels.find(qp.first) != labels.end()) {
+                matcher.emplace(qp.first, std::regex(qp.second.c_str()));
+            }
+        }
+        return (matcher.empty()) ? _false_function : [matcher](const mi::labels_type& labels) {
+            for (auto&& m : matcher) {
+                auto l = labels.find(m.first);
+                if (!std::regex_match((l == labels.end())? "" : l->second.c_str(), m.second)) {
+                    return true;
+                }
```

Comment:
So I see now that if you have multiple filters, the result is the OR of the different filters. This too needs a documentation in prometheus.md (and perhaps comment here in the code).

Reply:
It's actually AND (the filter function return true when it needs to filter), but it need to be document

Reply:
I don't understand... I see you returning "true" as soon as one match succeeded, and only returning false if all of them failed. So isn't it "OR"?

Or, maybe this filtering function is reversed in meaning? "true" means drop, not keep? If this is the case, please add comments (and/or consider reversing it back...)

Reply:
It's still super-confusing the way your "filter" function returns true for filtering the item **out**, not true for what should be kept. I think you must have at least a comment about that somewhere.

Reply:
I personally feel the reverse (filter function returns "true" for items that are to be kept) is clearer and more traditional (e.g., see https://en.wikipedia.org/wiki/Filter_(higher-order_function) or https://en.cppreference.com/w/cpp/ranges/filter_view), but if you really like the opposite one, please just add a comment warning future developers of this unconventional choice.

I would also rename the "false_function" to something which describes its real intention, i.e., "keep_everything" or something like that.